update-agent/core/src/components.rs

```diff
@@ -20,37 +25,109 @@ pub enum Location {
     Mcu,
 }
 
-#[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq, Hash)]
+#[derive(Serialize, Clone, Debug, PartialEq, Eq, Hash)]
 pub enum Device {
-    #[serde(rename = "emmc")]
-    Emmc,
-    #[serde(rename = "nvme")]
-    Nvme,
+    #[serde(rename = "ssd")]
+    Ssd,
     #[serde(rename = "qspi")]
     Qspi,
 }
 
+// Custom deserializer for Device that handles backward compatibility.
+// The manifest may contain "emmc" or "nvme" as device types, but they all
+// represent the same underlying SSD storage type. This deserializer transforms all
+// three values into Device::Ssd to unify the representation.
+impl<'de> Deserialize<'de> for Device {
+    fn deserialize<D>(deserializer: D) -> Result<Device, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct DeviceVisitor;
+
+        impl<'de> Visitor<'de> for DeviceVisitor {
+            type Value = Device;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a valid device type (emmc, nvme, qspi, ssd)")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Device, E>
+            where
+                E: de::Error,
+            {
+                match value {
+                    "emmc" | "nvme" | "ssd" => Ok(Device::Ssd),
+                    "qspi" => Ok(Device::Qspi),
+                    _ => Err(de::Error::unknown_variant(
+                        value,
+                        &["emmc", "nvme", "qspi", "ssd"],
+                    )),
+                }
+            }
+        }
+
+        deserializer.deserialize_str(DeviceVisitor)
+    }
+}
+
+/// Finds the block device where the given mountpoint is mounted.
+///
+/// This function uses stat to get the device major and minor numbers of the mountpoint,
+/// then determines the parent block device.
+///
+/// That function is an eqivalent of this bash script
+///
+/// dev=$(stat -c '%d' <mountpoint>)
+/// major=$((dev / 256))
+/// minor=$((dev % 256))
+/// sysfs_path="/sys/dev/block/$major:$minor"
+/// link_target=$(readlink -f $sysfs_path)
+/// device_name=$(basename $(dirname $link_target))
+/// echo "/dev/$device_name"
+fn find_block_device_by_mountpoint(
+    mountpoint: &std::path::Path,
+) -> std::io::Result<PathBuf> {
+    // 'stat' the mountpoint. (see man 2 stat)
+    let metadata = std::fs::metadata(mountpoint)?;
+
+    // Get major & minor of the underlying device
+    let dev = metadata.dev();
+    let major = dev >> 8;
```

Comment:
use https://docs.rs/rustix/latest/rustix/fs/fn.major.html and https://docs.rs/rustix/latest/rustix/fs/fn.minor.html.

Yes, I know its a dependency for only 1 thing that could be a bitwise operation. But in my defense, the linux docs say we should use the functions too. https://www.man7.org/linux/man-pages/man3/major.3.html